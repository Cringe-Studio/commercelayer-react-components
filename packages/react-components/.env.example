# Unit tests
VITE_TEST_CLIENT_ID=
VITE_TEST_ENDPOINT=
VITE_TEST_MARKET_ID=
VITE_TEST_USERNAME=
VITE_TEST_PASSWORD=
VITE_TEST_CLIENT_ID_INTEGRATION=
VITE_TEST_CLIENT_SECRET=
