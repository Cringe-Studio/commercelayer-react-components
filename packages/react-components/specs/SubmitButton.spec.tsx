import React from 'react'
import { SubmitButton } from '../packages/react-components/src'
import renderer from 'react-test-renderer'
import components from '../#config/components'

const propTypes = components.SubmitButton.propTypes

test('<SubmitButton/>', () => {
  expect.assertions(3)
  const component = renderer.create(<SubmitButton />)
  const tree = component.toJSON()
  const root = component.toTree()
  const proptypes = root.type['propTypes']
  expect(tree).toMatchSnapshot()
  expect(proptypes.children).toBe(propTypes.children)
  expect(proptypes.label).toBe(propTypes.label)
})

test('<SubmitButton with props />', () => {
  expect.assertions(3)
  const component = renderer.create(<SubmitButton />)
  const tree = component.toJSON()
  const root = component.toTree()
  const rendered = root.rendered
  expect(tree).toMatchSnapshot()
  expect(rendered.nodeType).toBe('host')
  expect(rendered.type).toBe('button')
})

test('<SubmitButton with custom children />', () => {
  expect.assertions(5)
  const CustomComponent = (props) => <span>{props.label}</span>
  const component = renderer.create(
    <SubmitButton label="My checkout">{CustomComponent}</SubmitButton>
  )
  const tree = component.toJSON()
  const root = component.toTree()
  const rendered = root.rendered
  const childRendered = root.rendered.rendered
  expect(tree).toMatchSnapshot()

  expect(rendered.props.children).toBe(CustomComponent)

  expect(childRendered.nodeType).toBe('component')
  expect(childRendered.type).toBe(CustomComponent)

  expect(childRendered.props.label).toBe('My checkout')
})
