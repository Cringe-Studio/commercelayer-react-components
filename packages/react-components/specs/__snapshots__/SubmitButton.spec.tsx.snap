// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SubmitButton with custom children /> 1`] = `
<span>
  My checkout
</span>
`;

exports[`<SubmitButton with props /> 1`] = `
<button
  type="submit"
>
  Submit
</button>
`;

exports[`<SubmitButton/> 1`] = `
<button
  type="submit"
>
  Submit
</button>
`;
